torch/distributed/distributed_c10d.py

```diff
@@ -2082,6 +2082,11 @@ def all_gather(tensor_list, tensor, group=None, async_op=False):
         [tensor([1.+1.j, 2.+2.j]), tensor([3.+3.j, 4.+4.j])] # Rank 1
 
     """
+    # `tensor_list` would be understood as "tensor or list."
```

Comment:
maybe then best name it more verbosely `tensor_or_list` to avoid understanding as "list of tensors"?

Reply:
Thanks for the suggestion. I changed the name to `output` to be generic (also to indicate it is an output).
In the documentation, I added that `output` can be a tensor or a list of tensor.